terraform/example/main.tf

```diff
@@ -1,38 +1,235 @@
-    root_ca_path = "/Users/xnutsive/go/src/github.com/gravitational/teleport-plugins/docker/tmp/auth.cas"
+  addr         = "localhost:3025"
+  cert_path    = var.cert_path
+  key_path     = var.key_path
+  root_ca_path = var.root_ca_path
+
+  # addr = "evilmartians.teleport.sh:443"
+  # identity_file_path = "id"
 }
 
-resource "teleport_user" "tf_test" {
+resource "teleport_provision_token" "example" {
+  metadata {
+    name = "example"
+    expires = "2022-10-12T07:20:50.5Z"
+    description = "Example token"
+
+    labels = {
+      example = "yes" 
+    }
+  }
 
+  spec {
+    roles = ["Node", "Auth"]
+  }
+}
+
+resource "teleport_user" "example" {
   metadata {
-    name = "Nate"
-    description = "Terraform Test User"
+    name        = "example"
+    description = "Example Teleport User"
+
+    expires = "2022-10-12T07:20:50.5Z"
+
+    labels = {
+      example = "yes"
+    }
+  }
+
+  spec {
+    roles = ["example"]
+
+    oidc_identities {
+      connector_id = "ocid1"
+      username     = "example"
+    }
+
+    oidc_identities {
+      connector_id = "ocid2"
+      username     = "example"
+    }
+
+    traits {
+      key   = "logins1"
+      value = ["example"]
+    }
+
+    traits {
+      key   = "logins2"
+      value = ["example"]
+    }
 
-    # This will probably explode
-    expires = "2022-10-12T07:20:50.52Z"
+    github_identities {
+      connector_id = "github"
+      username     = "example"
+    }
+  }
+}
 
+resource "teleport_role" "example" {
+  metadata {
+    name        = "example"
+    description = "Example Teleport Role"
+    expires     = "2022-10-12T07:20:50.52Z"
     labels = {
-      test = "label value"
+      example  = "yes"      
     }
   }
+  
+  spec {
+    options {
+      forward_agent           = false
+      max_session_ttl         = "7m"
+      port_forwarding         = false
+      client_idle_timeout     = "1h"
+      disconnect_expired_cert = true
+      permit_x11forwarding    = false
+      request_access          = "denied"
+    }
 
+    allow {
+      logins = ["example"]
+
+      rules {
+        resources = ["user", "role"]
+        verbs = ["list"]
+      }
+
+      request {
+        roles = ["example"]
+        claims_to_roles {
+          claim = "example"
+          value = "example"
+          roles = ["example"]
+        }
+      }
+
+      node_labels {
+         key = "example"
+         value = ["yes"]
+      }
+    }
+
+    deny {
+      logins = ["anonymous"]
+    }
+  }
+}
+
+resource "teleport_github_connector" "github" {
+  metadata {
+     name = "test"
+     labels = {
+       test = "yes"
+     }
+  }
   spec {
-    roles = ["foo"]
+    client_id = "gzigzigzeo"
+    client_secret = "value"
 
-    # TODO add Traits and Identities.
-    # Traits schema is fucked.
+    teams_to_logins {
+       organization = "gravitational"
+       team = "em"
+       logins = ["terraform"]
+    }
   }
+}
+
+# resource "teleport_trusted_cluster" "cluster" {
```

Comment:
Does this code have to be commented? I would either uncomment it, or remove.

Reply:
This resource won't work just from scratch, without additional setup. I've provided template configuration here. If you think, that we should remove it and user will pick it up from the docs later - let's do it.